### PR TITLE
Correctly locate Clang [Fixes 250]

### PR DESCRIPTION
**Pull request by shahid-pk**
Proposed for #250 and #243 Clang locating issues.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -57,7 +57,13 @@ check_prereqs()
     hash cmake 2>/dev/null || { echo >&2 "Please install cmake before running this script"; exit 1; }
     
     # Check for clang
-    hash clang 2>/dev/null || { echo >&2 "Please install clang before running this script"; exit 1; }
+    OS=`uname`
+    if [ $OS = "Linux" ]; then
```
Comment (janvorli) [inline build.sh]: The clang naming is not a general Linux problem. It is interesting that e.g. in my Ubuntu, all the llvm-xxx commands have the 3.5 suffix, but the clang doesn't.

Comment (shahid-pk) [inline build.sh]: so what would you suggest we do here ? check for both clang and clang-* for linux?.

Comment (janvorli) [inline build.sh]: Yes, it seems like we need to do it that way.

File: src/pal/tools/setup-compiler-clang.sh [hunk 2]
```diff
@@ -3,5 +3,22 @@
 # This file sets the environment to be used for building with clang.
 #
 
-export CC="$(which clang)"
-export CXX="$(which clang++)"
+OS=`uname`
+desired_clang_version=3.5
+
+locate_clang() {
```
Comment (janvorli) [inline src/pal/tools/setup-compiler-clang.sh]: We should also have a fallback here. First try to find the clang-xxx, if it is not found, then try the plain clang.

File: build.sh (hunk 1 above)
Comment (shahid-pk) [inline build.sh]: ok

**Comment by shahid-pk** [comment]
@janvorli  comments addressed .

**Comment by shahid-pk** [comment]
extra commit squashed

**Comment by janvorli** [comment]
LGTM, thanks for fixing it.

**Comment by mmitche** [comment]
LGTM

**Pull request merged**